packages/flutter_tools/lib/src/globals_null_migrated.dart

```diff
@@ -0,0 +1,161 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:process/process.dart';
+
+import 'base/config.dart';
+import 'base/context.dart';
+import 'base/error_handling_io.dart';
+import 'base/file_system.dart';
+import 'base/io.dart';
+import 'base/logger.dart';
+import 'base/net.dart';
+import 'base/os.dart';
+import 'base/platform.dart';
+import 'base/process.dart';
+import 'base/signals.dart';
+import 'base/template.dart';
+import 'base/terminal.dart';
+import 'base/time.dart';
+import 'base/user_messages.dart';
+import 'ios/plist_parser.dart';
+
+Config get config => context.get<Config>()!;
+HttpClientFactory get httpClientFactory => context.get<HttpClientFactory>()!;
+Logger get logger => context.get<Logger>()!;
+OperatingSystemUtils get os => context.get<OperatingSystemUtils>()!;
+Signals get signals => context.get<Signals>() ?? LocalSignals.instance;
+
+/// Currently active implementation of the file system.
+///
+/// By default it uses local disk-based implementation. Override this in tests
+/// with [MemoryFileSystem].
+FileSystem get fs => ErrorHandlingFileSystem(
+  delegate: context.get<FileSystem>() ?? localFileSystem,
+  platform: platform,
+);
+
+FileSystemUtils get fsUtils => context.get<FileSystemUtils>() ?? FileSystemUtils(
+  fileSystem: fs,
+  platform: platform,
+);
+
+const ProcessManager _kLocalProcessManager = LocalProcessManager();
+
+/// The active process manager.
+ProcessManager get processManager => context.get<ProcessManager>() ?? _kLocalProcessManager;
+ProcessUtils get processUtils => context.get<ProcessUtils>()!;
+
+const Platform _kLocalPlatform = LocalPlatform();
+Platform get platform => context.get<Platform>() ?? _kLocalPlatform;
+
+UserMessages get userMessages => context.get<UserMessages>()!;
+
+final OutputPreferences _default = OutputPreferences(
+  wrapText: stdio.hasTerminal,
+  showColor:  platform.stdoutSupportsAnsi,
+  stdio: stdio,
+);
+OutputPreferences get outputPreferences => context.get<OutputPreferences>() ?? _default;
+
+/// The current system clock instance.
+SystemClock get systemClock => context.get<SystemClock>() ?? _systemClock;
+SystemClock _systemClock = const SystemClock();
```

Comment:
This is the only one I added a fallback to for g3, which seems safe enough.  The rest either already have fallbacks or are `!`.